TypewiseAlert.Test/TypewiseAlertTest.cs

```diff
@@ -1,15 +1,138 @@
 using System;
 using Xunit;
+using TypewiseAlert.Notifiers;
+using static TypewiseAlert.TypewiseAlert;
 
 namespace TypewiseAlert.Test
 {
   public class TypewiseAlertTest
   {
+
+    [Fact]
+    public void InfersBreachForLowLimit()
+    {
+        Assert.True(InferBreach(12, 20, 30) == BreachType.TOO_LOW);
+    }
+
+    [Fact]
+    public void InfersBreachForHighLimit()
+    {
+        Assert.True(InferBreach(35, 20, 30) == BreachType.TOO_HIGH);
+    }
+
     [Fact]
-    public void InfersBreachAsPerLimits()
+    public void InfersBreachForNormal()
     {
-      Assert.True(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+        Assert.True(InferBreach(22, 20, 30) == BreachType.NORMAL);
     }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForPassiveCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -5) == BreachType.TOO_LOW);
+    }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForMediumCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 45) == BreachType.TOO_HIGH);
+    }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForHighCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 42) == BreachType.NORMAL);
+    }
+
+    [Fact]
+    public void EmailNotificationTest()
```

Comment:
invalid behavioral testing .....looks like it just a state based testing

Reply:
Implemented behavioral test cases for CheckAndAlert methods for all cases i.e., Console, Controller, Email. I have implemented Composite Pattern to send alert to all types of notifiers.